compiler/env/OMRCPU.cpp

```diff
@@ -23,19 +23,34 @@
 #pragma csect(STATIC,"OMRCPUBase#S")
 #pragma csect(TEST,"OMRCPUBase#T")
 
+#include <string.h>
 #include "env/CPU.hpp"
 #include "env/CompilerEnv.hpp"
-#include "infra/Assert.hpp"
 #include "env/defines.h"
+#include "infra/Assert.hpp"
 #include "omrcfg.h"
+#include "omrformatconsts.h"
+
+OMR::CPU::CPU():
+   _processor(TR_NullProcessor),
+   _endianness(TR::endian_unknown),
+   _majorArch(TR::arch_unknown),
+   _minorArch(TR::m_arch_none)
+   {
+   if (TR::Compiler->omrPortLib)
+      {
+      OMRPORT_ACCESS_FROM_OMRPORT(TR::Compiler->omrPortLib);
+      omrsysinfo_get_processor_description(&_processorDescription);
+      memset(_featureMasks, ~0, OMRPORT_SYSINFO_FEATURES_SIZE*sizeof(uint32_t));
```

Comment:
Is the intention to initialize all bits to 1? If so, then `-1` is a safer way of expressing that.
```suggestion
      memset(_featureMasks, -1, OMRPORT_SYSINFO_FEATURES_SIZE*sizeof(uint32_t));
```

Also, I'm not sure I understand why this is needed. Is `_featureMasks` supposed to show what feature bits are initialized?

Reply:
`_featureMasks` is used to show what feature bits the compiler cares about (so by default in the common CPU we have all 1's). Note that this is not needed for the `supportsFeature(feature)` type of queries. The purpose of this is to mask out all the "don't care" bits to make it easier for compatibility checks among cpu's. I don't think this design is finalized yet. It's what we currently use for x86 that I just want to port over for now.